packages/fast-components-styles-msft/src/utilities/density.ts

```diff
@@ -1,6 +1,146 @@
+        return typeof unit === "string" ? `${value}${unit}` : toPx(value);
+    };
+}
+
+/**
+ * Returns the component height as a number.
+ *
+ * @param lines The logical number of lines the component takes, typically 1.
+ * @param minusBorder Any border that should be removed from the overall height.
+ */
+export function heightNumber(
+    lines: number,
+    minusBorder: number = 0
+): (config: DesignSystem) => number {
+    return (config: DesignSystem): number => {
+        const designSystem: DesignSystem = withDesignSystemDefaults(config);
+        const value: number =
+            (designSystem.defaultHeightMultiplier + designSystem.density) *
+                designSystem.designUnit *
+                lines -
+            minusBorder;
+        return value;
+    };
+}
+
+export function minHeight(
+    lines: number = 1,
+    minusBorder: number = 0
+): (config: DesignSystem) => string {
+    return (config: DesignSystem): string => {
+        const designSystem: DesignSystem = withDesignSystemDefaults(config);
+        const value: number =
+            (designSystem.defaultHeightMultiplier - 3) * designSystem.designUnit * lines -
+            minusBorder;
+        return toPx(value);
+    };
+}
+
+export function maxHeight(
+    lines: number = 1,
+    minusBorder: number = 0
+): (config: DesignSystem) => string {
+    return (config: DesignSystem): string => {
+        const designSystem: DesignSystem = withDesignSystemDefaults(config);
+        const value: number =
+            (designSystem.defaultHeightMultiplier + 3) * designSystem.designUnit * lines -
+            minusBorder;
+        return toPx(value);
+    };
+}
+
+/**
+ * Returns the standard horizontal padding for text and icons formatted in the provided unit or px by default.
+ *
+ * @param minusBorder Any border that should be removed from the overall content padding.
+ * @param unit The unit of measurement; px by default.
+ */
+export function padding(
+    minusBorder: number = 0,
+    unit?: string
+): (config: DesignSystem) => string {
+    return (config: DesignSystem): string => {
+        const designSystem: DesignSystem = withDesignSystemDefaults(config);
+        const offset: number = densityToTypeOffset(designSystem);
+        const value: number = paddingNumber(minusBorder)(designSystem);
+        return typeof unit === "string" ? `${value}${unit}` : toPx(value);
+    };
+}
+
+/**
+ * Returns the standard horizontal padding for text and icons as a number.
+ *
+ * @param minusBorder Any border that should be removed from the overall content padding.
+ * @param unit The unit of measurement; px by default.
+ */
+export function paddingNumber(minusBorder: number = 0): (config: DesignSystem) => number {
+    return (config: DesignSystem): number => {
+        const designSystem: DesignSystem = withDesignSystemDefaults(config);
+        const offset: number = densityToTypeOffset(designSystem);
```

Comment:
it seems a bit odd to me to be using a type offset scale to derive this -  how is this related to typography?

Reply:
Because the horizontal padding is related to how large the text is. We only have three text sizes, and only adjust the padding when the text size changes.

Reply:
Modified per our conversation.